### PR TITLE
For code review of first set of fixes

### PR DESCRIPTION
**Pull request by pulinagrawal**
for code review

### CODE REVIEW

File: src/agent.py [hunk 1]
```diff
@@ -4,17 +4,24 @@
...
 import gym_tictactoe  # Needed to add 'TicTacToe-v0' into gym registry
 
+import logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
 # Number of cognitive cycles to execute (None -> forever)
-N_STEPS = None
+N_STEPS = 1000
+
+experimenting = True
```
Comment (pulinagrawal) [inline src/agent.py]: add comment about changing it to logger once we figure out logging

File: src/agent.py [hunk 2]
```diff
@@ -31,13 +38,11 @@ def lambda_mark_detector(mark, pos):
...
+pam = PerceptualAssociativeMemory(initial_concepts=[
                                                     #TODO: affective_valence is really valence
-                                                    "happy": CognitiveContent("happy", affective_valence=1.0),
-                                                    "sad": CognitiveContent("sad", affective_valence=-1.0),
-                                                    "win": CognitiveContent("win"),
-                                                    "lose": CognitiveContent("lose"),
-                                                    "draw": CognitiveContent("draw")})
+                                                    FeelingNode("happy", affective_valence=1.0),
+                                                    FeelingNode("sad", affective_valence=-1.0),
+                                                    ])
```
Comment (pulinagrawal) [inline src/agent.py]: Use these as interpretive feeling nodes.

Comment (pulinagrawal) [reply src/agent.py]: Interprets a positive reward signal from the environment as "happy" feeling and a negative reward signal as "sad" feeling.

File: src/common.py [hunk 3]
```diff
@@ -47,6 +64,7 @@ def receive_sensors(self, sensors):
             sensory_scene_content.append(fd.apply(sensors))
```
Comment (pulinagrawal) [inline src/common.py]: Feature Detectors activate content in PAM.

File: src/agent.py [hunk 4]
```diff
@@ -62,31 +67,42 @@ def lambda_mark_detector(mark, pos):
...
-broadcast_recipients = []
+broadcast_recipients = [procedural_memory]
 
 # Initialize Cueing Process
 cue_process = CueingProcess(cueable_modules)
 coalition_manager = CoalitionManager()
 
-removal_activation = {CurrentSituationalModel: CognitiveContent.current_activation,
-                      GlobalWorkspace: Coalition.activation
+removal_activation = {CurrentSituationalModel: 'current_activation',
```
Comment (pulinagrawal) [inline src/agent.py]: Make decay a module specific concern and put implementation within each module.

File: src/common.py [hunk 5]
```diff
@@ -374,10 +525,38 @@ def candidate_behaviors(self):
...
+
+            #Learn
+            '''
+            Learn([scheme], function=math.tanh,
+                  factor=scheme.current_activation*match_pct(scheme.result, broadcast.content))
+            '''
+
+            #Duplicate
+
+            if scheme.result is None or match_pct(scheme.result, broadcast.content)<1.0:
```
Comment (pulinagrawal) [inline src/common.py]: From recently selected behavior

File: src/agent.py [hunk 6]
```diff
@@ -117,45 +134,56 @@ def run(environment, n=None, render=True):
...
+            candidate_behaviors = procedural_memory.candidate_behaviors
+            action_selection.receive_behaviors(candidate_behaviors)
 
             # Process selected behavior
             selected_behavior = action_selection.selected_behavior
             if selected_behavior is not None:
                 # Expectation codelet created from selected behavior
-                attn_codelets.append(AttentionCodelet(select=lambda x: x == selected_behavior.result))
+                if selected_behavior.result is None:
+                    attn_codelets.append(ExpectationCodelet(scheme=selected_behavior,
```
Comment (pulinagrawal) [inline src/agent.py]: Remove this scheme for the updated procedural memory learning from recently selected behaviors.

File: src/common.py [hunk 7]
```diff
@@ -162,6 +178,21 @@ def __str__(self):
     def __eq__(self, other):
         return self.content == other.content if hasattr(other, "content") else False
 
+    def __hash__(self):
+        return hash(self.content)
+
+
+class FeelingNode(CognitiveContent):
```
Comment (pulinagrawal) [inline src/common.py]: For now interpretive feeling node

File: src/common.py [hunk 8]
```diff
@@ -184,13 +184,17 @@ def __hash__(self):
 
 class FeelingNode(CognitiveContent):
 
-    def __init__(self, content, affective_valence=0.0, current_activation=0.0, bla=0.0):
+    def __init__(self, content, valence=0.0, current_activation=0.0, bla=0.0):
```
Comment (skugele) [inline src/common.py]: Why "valence" instead of "valence_sign"?

**Pull request merged**